### PR TITLE
Dns z bit 4515

### PR DESCRIPTION
**Pull request by odinje**
Make sure these boxes are signed before submitting your Pull Request -- thank you.

- [x] I have read the contributing guide lines at https://redmine.openinfosecfoundation.org/projects/suricata/wiki/Contributing
- [x] I have signed the Open Information Security Foundation contribution agreement at https://suricata-ids.org/about/contribution-agreement/
- [x] I have updated the user guide (in doc/userguide/) to reflect the changes made (if applicable)

Link to [Feature #4515](https://redmine.openinfosecfoundation.org/issues/4515) ticket:

Describe changes:
- Adding logging of Z-bit in DNS logs
- Updating eve-json-format documentation for DNS to include this bit

#suricata-verify-pr:
#suricata-verify-repo:
#suricata-verify-branch:
#suricata-update-pr:
#suricata-update-repo:
#suricata-update-branch:
#libhtp-pr:
#libhtp-repo:
#libhtp-branch:

### CODE REVIEW

File: rust/src/dns/log.rs [hunk 1]
```diff
@@ -503,6 +503,7 @@ fn dns_log_json_answer(js: &mut JsonBuilder, response: &DNSResponse, flags: u64)
     if header.flags & 0x0080 != 0 {
         js.set_bool("ra", true)?;
     }
+    js.set_uint("z", ((header.flags & 0x0070) >> 4).into());
```
Comment (jasonish) [inline rust/src/dns/log.rs:506]: Why int when bool is used for the other flags?

Comment (odinje) [reply rust/src/dns/log.rs:506]: @jasonish The answer is that used the same method that was used in Zeeks parser: https://github.com/zeek/zeek/blob/master/src/analyzer/protocol/dns/DNS.cc
They are also using bools for the other fields, but an int for this one.

Comment (jasonish) [reply rust/src/dns/log.rs:506]: Yeah, I think bool makes more sense. Fits better with how we log the other flags.

**Comment by odinje** [comment]
Update in #6187

**Pull request closed without merging**